File: golang/_0026_Remove_Duplicates_from_Sorted_Array/main.go
package main

func removeDuplicates(nums []int) int {
	n, i := len(nums), 0
	if n <= 1 {
		return n
	}

	for j := 1; j < n; j++ {
		if nums[j] > nums[i] {
			i++
			nums[i] = nums[j]
		}
	}

	return i + 1
}

File: python3/_0026_Remove_Duplicates_from_Sorted_Array/main.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums:
            return 0

        i, j = 0, 1
        while j < len(nums):
            if nums[j] > nums[i]:
                nums[i + 1] = nums[j]
                i += 1
            j += 1

        return i + 1
